jwt.js

```diff
@@ -175,7 +179,11 @@ function fastifyJwt (fastify, options, next) {
 
     steed.waterfall([
       function getSecret (callback) {
-        secretCallbackSign(reply.request, payload, callback)
+        const signResult = secretCallbackSign(reply.request, payload, callback)
+
+        if (signResult && typeof signResult.then === 'function') {
+          signResult.then(result => callback(null, result), callback)
```

Comment:
What about rejection?

Reply:
is it not handled?

Reply:
Ah! I see you're using the second parameter of `then` instead of `catch`.